src/plugins/data/public/ui/search_bar/search_bar.tsx

```diff
@@ -371,6 +373,14 @@ class SearchBarUI extends Component<SearchBarProps, State> {
         }
       }
     );
+    const dataSet = this.dataSetService.getDataSet();
```

Comment:
I think no other plugin utilizes datasets besides Discover. However, should we consider this while designing this recent query and history. Specifically the app where the query was sent. so that if the current app path is not the same app path then we do not display the query. in case one app supports different languages but another doesn't